<?xml version="1.0"?>
<launch>

  <param name="/moveit_planner/fake_execution" value="true"/>
  <node pkg="moma_mission" type="mock_modules.py"  name="mock_modules" output="screen">
    <param name="gripper_action_topic"          value="/my_gen3/arm_robotiq_2f_85_gripper_controller/gripper_cmd"/>
    <param name="gripper_command_topic"          value="/gripper_cmd"/>
    <param name="base_odom_topic"               value="/base_pose_measured"/>
    <param name="nav_goal_topic"                value="/move_base_simple/goal"/>
    <param name="ee_goal_topic"                 value="/target_pose"/>
    <param name="ee_path_topic"                 value="/desired_path"/>
    <param name="detection_topic"               value="/object_keypoints_ros/pose"/>
    <param name="joint_action_topic"            value="/joint_action"/>
    <param name="roco_manager_namespace"        value="/smb_highlevel_controller"/>
    <param name="ros_control_manager_namespace" value="/my_gen3"/>
  </node>

  <!--    Transforms needed for the state machine to work -->
  <node name="world_arm_base_link_bc"  type="static_transform_publisher"  pkg="tf2_ros"
        args="0 0 0 0 0 0 world arm_base_link"/>

  <node name="world_camera"  type="static_transform_publisher"  pkg="tf2_ros"
        args="1.0 1.0 1.0 0 0 0 world camera"/>

  <node name="world_map"  type="static_transform_publisher"  pkg="tf2_ros"
          args="0 0 0 0 0 0 world map"/>

  <node name="world_navigation_target"  type="static_transform_publisher"  pkg="tf2_ros"
          args="0 0 0 0 0 0 world navigation_target"/>

<!--  TODO it cannot retrieve the static tf publish during the detection state fix this-->
  <node name="world_valve_frame"  type="static_transform_publisher"  pkg="tf2_ros"
          args="1.0 1.0 0.2 0 0 0 world valve_frame"/>

</launch>
